osu.Game/Screens/Ranking/Statistics/HitEventTimingDistributionGraph.cs

```diff
@@ -79,105 +104,174 @@ private void load()
+                    float alpha = 1f - position * 0.8f;
+
+                    axisFlow.Add(new OsuSpriteText
+                    {
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        RelativePositionAxes = Axes.X,
+                        X = -position / 2,
+                        Alpha = alpha,
+                        Text = axisValue.ToString("-0"),
+                        Font = OsuFont.GetFont(size: axis_font_size, weight: FontWeight.SemiBold)
+                    });
+
+                    axisFlow.Add(new OsuSpriteText
+                    {
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        RelativePositionAxes = Axes.X,
+                        X = position / 2,
+                        Alpha = alpha,
+                        Text = axisValue.ToString("+0"),
+                        Font = OsuFont.GetFont(size: axis_font_size, weight: FontWeight.SemiBold)
+                    });
+                }
             }
         }
 
         private class Bar : CompositeDrawable
         {
-            public Bar()
+            private readonly float value;
+            private readonly float maxValue;
+
+            private readonly Circle boxOriginal;
+            private Circle boxAdjustment;
+
+            private const float minimum_height = 0.05f;
+
+            public Bar(float value, float maxValue, bool isCentre)
             {
-                Anchor = Anchor.BottomCentre;
-                Origin = Anchor.BottomCentre;
+                this.value = value;
+                this.maxValue = maxValue;
 
                 RelativeSizeAxes = Axes.Both;
+                Masking = true;
 
-                InternalChild = new Circle
+                InternalChildren = new Drawable[]
                 {
-                    RelativeSizeAxes = Axes.Both,
-                    Colour = Color4Extensions.FromHex("#66FFCC")
+                    boxOriginal = new Circle
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                        Anchor = Anchor.BottomCentre,
+                        Origin = Anchor.BottomCentre,
+                        Colour = isCentre ? Color4.White : Color4Extensions.FromHex("#66FFCC"),
+                        Height = minimum_height,
+                    },
                 };
             }
+
+            private const double duration = 300;
+
+            protected override void LoadComplete()
+            {
+                base.LoadComplete();
+
+                float height = Math.Clamp(value / maxValue, minimum_height, 1);
+
+                if (height > minimum_height)
+                    boxOriginal.ResizeHeightTo(height, duration, Easing.OutQuint);
+            }
+
+            public void UpdateOffset(float adjustment)
+            {
+                bool hasAdjustment = adjustment != value;
```

Comment:
I would make this say

```suggestion
                bool hasAdjustment = adjustment != value && adjustment / maxValue >= minimum_height;
```

Rationale is that without that latter check, little yellow dots appear at the bottom in some columns and look a bit wrong:

![2022-03-05-161459_709x213_scrot](https://user-images.githubusercontent.com/20418176/156889804-030e7284-a88e-48ed-977e-7e347afc7513.png)

Compare with how this looks after applying:

![2022-03-05-162918_623x219_scrot](https://user-images.githubusercontent.com/20418176/156889841-084fe236-e384-4a70-88d6-12d70c02bd09.png)